contributors/design-proposals/aggregated-api-servers.md

```diff
@@ -0,0 +1,266 @@
+  aggregated servers until they become stable, at which point, they can be moved
+  to the core kubernetes master, if appropriate.
+* Ensure that new APIs follow kubernetes conventions: Without the mechanism
+  proposed here, community members might be forced to roll their own thing which
+  may or may not follow kubernetes conventions.
+
+## Goal
+
+* Developers should be able to write their own API server and cluster admins
+  should be able to add them to their cluster, exposing new APIs at runtime. All
+  of this should not require any change to the core kubernetes API server.
+* These new APIs should be seamless extension of the core kubernetes APIs (ex:
+  they should be operated upon via kubectl).
+
+## Non Goals
+
+The following are related but are not the goals of this specific proposal:
+* Make it easy to write a kubernetes API server.
+
+## High Level Architecture
+
+There will be a new component in the cluster, `kube-aggregator`, which has these
+responsibilities:
+* Provide an API for registering API servers.
+* Summarize discovery information from all the servers.
+* Proxy client requests to individual servers.
+
+The reverse proxy is provided for convenience. Clients can discover server URLs
+using the summarized discovery information and contact them directly. Simple
+clients can always use the proxy and don't need to know that under the hood
+multiple apiservers are running.
+
+Wording note: When we say "API servers" we really mean groups of apiservers,
+since any individual apiserver is horizontally replicatable. Similarly,
+kube-aggregator itself is horizontally replicatable.
+
+## Operational configurations
+
+There are two configurations in which it makes sense to run `kube-aggregator`.
+ 1. In test or single-user mode. An individual developer who wants to test
+    their own apiserver could run their own private copy of `kube-aggregator`,
+    configured such that only they can interact with it. This allows for testing
+    both `kube-aggregator` and any custom apiservers without the potential for
+    causing any collateral damage in the rest of the cluster.
+ 2. In gateway mode. Each cluster should run kube-aggregator as the official
+    gateway to the cluster, where it aggregates all of the apiservers the cluster
+    administer wishes to provide.
+
+### Constraints
+
+* Unique API groups across servers: Each API server (and groups of servers, in HA)
+  should expose unique API groups.
+* Follow API conventions: APIs exposed by every API server should adhere to [kubernetes API
+  conventions](../devel/api-conventions.md).
+* Support discovery API: Each API server should support the kubernetes discovery API
+  (list the suported groupVersions at `/apis` and list the supported resources
+  at `/apis/<groupVersion>/`)
+* No bootstrap problem: The core kubernetes server should not depend on any
+  other aggregated server to come up. Other servers can only depend on the core
```

Comment:
"other servers can only depend on the core kubernetes server" seems overly strict.  They can only be *guaranteed* that the core kube server exists, but finding and using others doesn't seem like abuse as long as their absence is tolerated.